### ISSUE TITLE
unable to pass config file from pre-commit

### ISSUE THREAD

**Issue by 7adityaraj** [antonbabenko/pre-commit-terraform#225]
### Describe the bug

I looked over the below block on how tf docs pick the config file.
https://github.com/terraform-docs/terraform-docs/blob/3d84cad448378bc36dfef856a319ff6a43d7752f/internal/cli/run.go#L133

```
v.AddConfigPath(r.rootDir)              // first look at module root
	v.AddConfigPath(r.rootDir + "/.config") // then .config/ folder at module root
	v.AddConfigPath(".")                    // then current directory
	v.AddConfigPath(".config")              // then .config/ folder at current directory
	v.AddConfigPath("$HOME/.tfdocs.d")      // and finally $HOME/.tfdocs.d/

	if err := v.ReadInConfig(); err != nil {
		var perr *os.PathError
		if errors.As(err, &perr) {
			return fmt.Errorf("config file %s not found", config.File)
		}
```
I have my configuration files under **precommit** dir as below

```
user: Github/terraform-mod-ecs-service
└─ $ ls -la precommit/
 .
 ..
 .pre-commit-config.yaml
 .terraform-docs.yml
 Dockerfile
 docker-compose-multibranch.yml
 entrypoint.sh
 gitleaks-config.toml
```
my precommit config is:

```
#  pre-commit run --all-files --verbose
repos:
  - repo: https://github.com/antonbabenko/pre-commit-terraform
    rev: v1.50.0
    hooks:
      - id: terraform_docs
        args: [
          '--args=--sort-by required',
          '--args=--config=.terraform-docs.yml'
          #'--args=--config=precommit/.terraform-docs.yml'
        ]
        exclude: >
          (?x)^(
              examples/.*$
          )$
```

```
user: Github/terraform-mod-ecs-service
└─ $ pre-commit run -c ./precommit/.pre-commit-config.yaml --all-files --verbose
Terraform docs...........................................................Failed
- hook id: terraform_docs
- duration: 1.75s
- exit code: 1

Error: config file .terraform-docs.yml not found
```

```
user: terraform-mod-ecs-service/precommit
└─ $ pre-commit run -c .pre-commit-config.yaml --all-files --verbose
Terraform docs...........................................................Failed
- hook id: terraform_docs
- duration: 1.67s
- exit code: 1

Error: config file .terraform-docs.yml not found
```

### Please let us know what behaviour you expected and how terraform-docs diverged from that behaviour.

 .terraform-docs.yml should be picked as config for tf-docs or someone can suggest if there is other way.

Adding yml file to each directory where I have readme with tf-docs hooks works. But I would like to have only one yml file to the precommit dir if possible.

Thank you in advance.

### How can we reproduce it?
with above steps.

### Environment information

```
└─ $ pre-commit autoupdate
Updating https://github.com/antonbabenko/pre-commit-terraform ... already up to date.
└─ $ pre-commit --v
pre-commit 2.11.1

- terraform-docs version (use `terraform-docs --version`):  terraform-docs version v0.14.1 darwin/amd64
- Go version (if you manually built. use `go version`): go version go1.16.2 darwin/amd64
- OS (e.g. Windows, MacOS): Darwin 20.6.0 Darwin Kernel Version 20.6.0
```

**Comment by MaxymVlasov** [comment]
Hi, @7adityaraj .
I think it can be fixed by #155. You can try clone repo, checkout to PR branch, and use `pre-commit try-repo` ([more details](https://github.com/antonbabenko/pre-commit-terraform#run-and-debug-hooks-locally)) to check is it works for you.

After, check, please describe is it help you.

**Comment by 7adityaraj** [comment]
Hi @MaxymVlasov,

Thank you for the suggestions. 

I tried with commit ID as below, but it didn't work. May be I am still missing something

```
repos:
    - repo : https://github.com/jans-forks/pre-commit-terraform
      rev  : 39d811f
      hooks:
       - id: terraform_docs
         files: (\.tf)$
         exclude: >
          (?x)^(
              examples/.*$|
              modules/alb/.*$|
              modules/nlb/.*$|
              modules/elb/.*$|
              modules/elb-tcp/.*$|
              modules/simple/.*$|
              modules/alb-wi/.*$
          )$
         args:
          - --config=precommit/.terraform-docs.yml
          - md
```

```
>> running precommit from ./precommit/.pre-commit-config.yaml
Terraform docs...........................................................Failed
- hook id: terraform_docs
- duration: 1.47s
- exit code: 1

WORKING with: "--config=precommit/.terraform-docs.yml"
Adding arg
WORKING with: "md"
Adding arg
WORKING with: "modules/core/ecr.tf"
WORKING with: "config.tf"
WORKING with: "modules/core/locals.tf"
WORKING with: "main.tf"
WORKING with: "modules/core/task.tf"
WORKING with: "outputs.tf"
WORKING with: "modules/core/autoscaling.tf"
WORKING with: "modules/core/ecs_lb.tf"
WORKING with: "modules/core/data.tf"
WORKING with: "variables.tf"
WORKING with: "modules/core/outputs.tf"
WORKING with: "modules/core/variables.tf"
Error: config file precommit/.terraform-docs.yml not found

```

**Comment by MaxymVlasov** [comment]
Good, at least now it show right path to config file in error.

@7adityaraj , using this fork, try change first arg to

```yaml
- --config=__GIT_WORKING_DIR__/precommit/.terraform-docs.yml
```

**Comment by 7adityaraj** [comment]
Many thanks @MaxymVlasov,

It works when I pass the full path of the yml file. Also, I figure out with your help, it does work with the previous release tag 1.50.0 when I provide the full path of yml file. Is this expected or do we need to fill a bug for it.

**Comment by MaxymVlasov** [comment]
This is exepted behavior.

I will clarify docs about that.

**Comment by MaxymVlasov** [comment]
And `__GIT_WORKING_DIR__` - is not full path, it argument that should be passead as is

@7adityaraj

**Comment by MaxymVlasov** [comment]
Like this one
![Screenshot_20210915-002933.png](https://user-images.githubusercontent.com/11096782/133342287-c29f95dd-7862-4877-a0c5-595a35ce1ca8.png)

**Comment by 7adityaraj** [comment]
@MaxymVlasov,

It didn't work that way.

```
Terraform docs...........................................................Failed
- hook id: terraform_docs
- duration: 1.55s
- exit code: 1

WORKING with: "--config=__GIT_WORKING_DIR__/precommit/.terraform-docs.yml"
Adding arg
WORKING with: "md"
Adding arg
WORKING with: "modules/core/ecr.tf"
WORKING with: "config.tf"
WORKING with: "modules/core/locals.tf"
WORKING with: "main.tf"
WORKING with: "modules/core/task.tf"
WORKING with: "outputs.tf"
WORKING with: "modules/core/autoscaling.tf"
WORKING with: "modules/core/ecs_lb.tf"
WORKING with: "modules/core/data.tf"
WORKING with: "variables.tf"
WORKING with: "modules/core/outputs.tf"
WORKING with: "modules/core/variables.tf"
Error: config file __GIT_WORKING_DIR__/precommit/.terraform-docs.yml not found
```


```
Terraform docs...........................................................Passed
- hook id: terraform_docs
- duration: 2.75s

WORKING with: "--config=/Users/user1/Documents/GitHub/terraform-mod-ecs-service/precommit/.terraform-docs.yml"
Adding arg
WORKING with: "md"
Adding arg
WORKING with: "modules/core/ecr.tf"
WORKING with: "config.tf"
WORKING with: "modules/core/locals.tf"
WORKING with: "main.tf"
WORKING with: "modules/core/task.tf"
WORKING with: "outputs.tf"
WORKING with: "modules/core/autoscaling.tf"
WORKING with: "modules/core/ecs_lb.tf"
WORKING with: "modules/core/data.tf"
WORKING with: "variables.tf"
WORKING with: "modules/core/outputs.tf"
WORKING with: "modules/core/variables.tf"
```

**Comment by MaxymVlasov** [comment]
Is it work if use tag 1.50 from this repo?

```yaml
repos:
- repo: https://github.com/antonbabenko/pre-commit-terraform 
  rev: v1.50.0 
  hooks: 
  - id: terraform_docs
    args:
    - --config=__GIT_WORKING_DIR__/precommit/.terraform-docs.yml
    - md
```

**Comment by 7adityaraj** [comment]
no luck.

```
 - repo: https://github.com/antonbabenko/pre-commit-terraform
      rev: v1.50.0
      hooks:
       - id: terraform_docs
         args:
          - --config=__GIT_WORKING_DIR__/precommit/.terraform-docs.yml
          - md
         exclude: >
          (?x)^(
              examples/.*$|
              modules/alb/.*$|
              modules/nlb/.*$|
              modules/elb/.*$|
              modules/elb-tcp/.*$|
              modules/alb-wi/.*$|
              modules/alb/.*$|
              modules/simple/.*$
          )$
```

```
Terraform docs...........................................................Failed
- hook id: terraform_docs
- duration: 0.02s
- exit code: 1

getopt: unrecognized option '--config'
```

**Comment by MaxymVlasov** [comment]
That really looks like a bug.
Depends on #155 merge.

+ reminder: Need to document TFlint relative path usage 

Thank you, @7adityaraj !
I will fix this in ~3-4 weeks. If it too long - you can try prepare fix based on #155 and send PR (don't forget mention here if you'll start contibuting)

**Comment by 7adityaraj** [comment]
thank you @MaxymVlasov  and will wait for the next release. I am using a workaround as have a copy of yml file in each readme directory with tag 1.50.0

**Comment by MaxymVlasov** [comment]
>reminder: Need to document TFlint relative path usage

Already done

**Comment by tpdownes** [comment]
Why is this function doing such convolute stuff for functionality that is built-in to `terraform-docs`?

It basically does

```shell
terraform-docs markdown --config .tfdocs-markdown.yaml ${terraform_module_folder}
```

where the config file looks like:

```yaml
formatter: markdown

output:
  file: README.md
  mode: inject
  template: |-
    <!-- BEGINNING OF PRE-COMMIT-TERRAFORM DOCS HOOK -->
    {{ .Content }}
    <!-- END OF PRE-COMMIT-TERRAFORM DOCS HOOK -->
```

**Comment by antonbabenko** [comment]
@tpdownes In short, the terraform_docs hook in `pre-commit-terraform` used to handle a variety of versions of Terraform starting from versions before 0.12 (yes, it was a few years ago) and `terraform-docs` has been evolving.

We are improving it all the time.

**Comment by MaxymVlasov** [comment]
@7adityaraj you missing that it will be in 1.53.0 :)

You can you latest commit SHA from master branch, for tests proposes
